src/main/java/org/kairosdb/core/reporting/MetricReportingModule.java

```diff
@@ -76,6 +80,48 @@ public MetricReportingModule(final Properties properties) {
         this.properties = properties;
     }
 
+    static Map<String, Class<?>> parseTaggerBuilders(final Properties properties) {
+        final Map<String, Class<?>> taggerBuilders = Maps.newHashMap();
+        for (final String key : properties.stringPropertyNames()) {
+            if (key.startsWith(TAGGER_KEY_PREFIX) && key.endsWith(TAGGER_CLASS_SUFFIX)) {
+                final String taggerImplAsString = properties.getProperty(key);
+                try {
+                    @SuppressWarnings("unchecked") final Class<?> taggerBuilderClass = Class.forName(taggerImplAsString.trim() + "$Builder");
+                    taggerBuilders.put(
+                            key.substring(0, key.length() - TAGGER_CLASS_SUFFIX.length())
+                                    .substring(TAGGER_KEY_PREFIX.length()),
+                            taggerBuilderClass);
+                } catch (final ClassNotFoundException e) {
+                    throw new RuntimeException("Tagger class not found: " + taggerImplAsString, e);
+                }
+            }
+        }
+        return taggerBuilders;
+    }
+
+    static Map<String, Tagger> loadTaggers(final Map<String, Class<?>> taggerBuilders, final Properties properties) {
+        final JavaPropsSchema schema = new JavaPropsSchema()
+                .withFirstArrayOffset(0);
+        final Map<String, Tagger> taggers = Maps.newHashMap();
+        for (final Map.Entry<String, Class<?>> entry : taggerBuilders.entrySet()) {
+            final String taggerKey = entry.getKey();
+            final Class<?> builderClass = entry.getValue();
+            try {
+                final Object builder = PROPERTIES_MAPPER.readPropertiesAs(
+                        properties,
+                        schema.withPrefix(TAGGER_KEY_PREFIX + taggerKey),
+                        builderClass);
+
+                @SuppressWarnings("unchecked") final org.kairosdb.core.reporting.Tagger tagger =
```

Comment:
Do you need the fully qualified class name here?

Reply:
Fixed.